frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/pushdialog/GitPushDialog.java

```diff
@@ -61,42 +64,39 @@ public GitPushDialog(
         Registry.is("ide.perProjectModality") ? IdeModalityType.PROJECT : IdeModalityType.IDE);
     this.isForcePushRequired = isForcePushRequired;
     this.project = project;
-    this.pushAction = new PushSwingAction();
-    this.pushAction.putValue(DEFAULT_ACTION, Boolean.TRUE);
-
-    var dialog = new MyVcsPushDialog(project, selectedRepositories.asJava(), pushSource) {
-      @Override
-      @UIEffect
-      public @Nullable VcsPushOptionValue getAdditionalOptionValue(PushSupport support) {
-        return GitPushDialog.this.getAdditionalOptionValue(support);
-      }
-
-      @Override
-      @UIEffect
-      public JRootPane getRootPane() {
-        return GitPushDialog.this.getRootPane();
-      }
-    };
 
+    @SuppressWarnings("nullness:assignment.type.incompatible") @Initialized MyVcsPushDialog dialog = new MyVcsPushDialog(this,
```

Comment:
At least each place we stretch the type system is clearly visible now + range of warning suppressions is as narrow as possible